awx/main/analytics/collectors.py

```diff
@@ -0,0 +1,168 @@
+    counts['smart_inventories'] = models.Inventory.objects.filter(kind='smart').count()
+    counts['normal_inventories'] = models.Inventory.objects.filter(kind='').count()
+
+    active_sessions = Session.objects.filter(expire_date__gte=now()).count()
+    api_sessions = models.UserSessionMembership.objects.select_related('session').filter(session__expire_date__gte=now()).count()
+    channels_sessions = active_sessions - api_sessions
+    counts['active_sessions'] = active_sessions
+    counts['active_api_sessions'] = api_sessions
+    counts['active_channels_sessions'] = channels_sessions
+    counts['running_jobs'] = models.Job.objects.filter(status='running').count()
+    return counts
+    
+    
+@register('org_counts')
+def org_counts(since):
+    counts = {}
+    for org in models.Organization.objects.annotate(
+            num_users=Count('member_role__members', distinct=True), 
+            num_teams=Count('teams', distinct=True)):  # use .only()
+        counts[org.id] = {'name': org.name,
+                          'users': org.num_users,
+                          'teams': org.num_teams
+                          }
+    return counts
+    
+    
+@register('cred_type_counts')
+def cred_type_counts(since):
+    counts = {}
+    for cred_type in models.CredentialType.objects.annotate(
+            num_credentials=Count('credentials', distinct=True)):  
+        counts[cred_type.id] = {'name': cred_type.name,
+                                'credential_count': cred_type.num_credentials
+                                }
+    return counts
+    
+    
+@register('inventory_counts')
+def inventory_counts(since):
+    counts = {}
+    from django.db.models import Count
+    for inv in models.Inventory.objects.annotate(
+            num_sources=Count('inventory_sources', distinct=True), 
+            num_hosts=Count('hosts', distinct=True)).only('id', 'name', 'kind'):
+        counts[inv.id] = {'name': inv.name,
+                          'kind': inv.kind,
+                          'hosts': inv.num_hosts,
+                          'sources': inv.num_sources
+                          }
+    return counts
+
+
+@register('projects_by_scm_type')
+def projects_by_scm_type(since):
+    counts = dict(
+        (t[0] or 'manual', 0)
+        for t in models.Project.SCM_TYPE_CHOICES
+    )
+    for result in models.Project.objects.values('scm_type').annotate(
+        count=Count('scm_type')
+    ).order_by('scm_type'):
+        counts[result['scm_type'] or 'manual'] = result['count']
+    return counts
+
+
+@register('job_counts')   # TODO: evaluate if we want this (was not an ask)  Also, may think about annotating rather than grabbing objects for efficiency 
+def job_counts(since):    # TODO: Optimize -- for example, all of these are going to need to be restrained to the last 24 hours/INSIGHTS_SCHEDULE
+    counts = {}
+    counts['total_jobs'] = models.UnifiedJob.objects.all().count()
```

Comment:
Let's figure out how to express this all in the ORM as one query grouped by status i.e.,

```sql
awx=> SELECT COUNT(id), status FROM main_unifiedjob GROUP BY status;
 count |   status
-------+------------
   352 | failed
  2964 | successful
     2 | error
    10 | canceled
(4 rows)
```